pkg/sign/signer.go

```
@@ -15,13 +15,31 @@
 package sign
 
 import (
+	"encoding/pem"
+
 	protobundle "github.com/sigstore/protobuf-specs/gen/pb-go/bundle/v1"
 	protocommon "github.com/sigstore/protobuf-specs/gen/pb-go/common/v1"
 )
 
 const bundleV03MediaType = "application/vnd.dev.sigstore.bundle.v0.3+json"
 
-func Bundle(content Content, keypair Keypair, fulcio *Fulcio, idToken string, timestampAuthority *TimestampAuthority) (*protobundle.Bundle, error) {
+type BundleOptions struct {
+	// Optional Fulcio instance to get code signing certificate from.
+	//
+	// Resulting bundle will contain a certificate for its verification
+	// material content, instead of a public key.
+	Fulcio *Fulcio
+	// Optional OIDC JWT to send to Fulcio; required if using Fulcio
+	IDToken string
+	// Optional list of timestamp authorities to contact for inclusion in bundle
+	TimestampAuthorities []*TimestampAuthority
+	// Optional Rekor instance to get transparency log entry from.
+	//
+	// Supports hashedrekord and dsse entry types
+	Rekor *Rekor
+}
+
+func Bundle(content Content, keypair Keypair, opts BundleOptions) (*protobundle.Bundle, error) {
 	bundle := &protobundle.Bundle{MediaType: bundleV03MediaType}
 
 	// Sign content and add to bundle
```

Can we fail early if keypair == nil?